tests/app/submitter/test_submitter.py

```diff
@@ -212,7 +212,46 @@ def test_gcs_submitter_adds_metadata_when_sends_message(patch_client):
     }
 
 
-def test_gcs_feedback_submitter_uploads_feedback(patch_client):
+@pytest.mark.parametrize(
+    "submitter, entrypoint, data_to_upload",
+    [
+        (
+            GCSSubmitter,
+            "send_message",
+            {"message": "some message", "tx_id": "123", "case_id": "456"},
+        ),
+        (
+            GCSFeedbackSubmitter,
+            "upload",
+            {
+                "metadata": {"some-data": "some-value"},
+                "payload": json_dumps({"some-data": "some-value"}),
+            },
+        ),
+    ],
+)
+def test_gcs_submitter_retries_transient_errors(
+    patch_gcs_client, gcs_blob_with_retry, submitter, entrypoint, data_to_upload
+):
+    # Given
+    gcs_submitter = submitter(bucket_name="test_bucket")
+
+    # When
+    bucket = patch_gcs_client.return_value.get_bucket.return_value
+    bucket.blob.return_value = gcs_blob_with_retry
+
+    function_to_call = getattr(gcs_submitter, entrypoint)
+    successful = function_to_call(**data_to_upload)
+
+    # Then
+    assert (
+        gcs_blob_with_retry._get_transport().request.call_count  # pylint: disable=protected-access
```

Comment:
You could make this slightly clearer with the comments, its not immediately obvious why it gets called twice, unless you go back to the fixture and see you have a list of side_effects